pam/newpasswordmodel.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/charmbracelet/bubbles/textinput"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+var (
+	errorStyle = lipgloss.NewStyle().Foreground(lipgloss.Color("#ff0000"))
+)
+
+// newPasswordModel is the form layout type to allow authentication and return a challenge.
+type newPasswordModel struct {
+	errorMsg string
+	label    string
+
+	passwordEntries []*textinputModel
+	focusableModels []authenticationComponent
+	focusIndex      int
+}
+
+// newNewPasswordModel initializes and return a new newPasswordModel.
+func newNewPasswordModel(label, entryType, buttonLabel string) newPasswordModel {
+	var focusableModels []authenticationComponent
+	var passwordEntries []*textinputModel
+
+	// TODO: add digits and force validation.
```

Comment:
We should definitely create a tracking list for all the `//TODOS`

Reply:
we can easily do it with grep :)

Reply:
I know, but that implies that we'll remember to grep it :joy: